### ISSUE TITLE
No Sigmoid activation in final layer of Generator?

### ISSUE THREAD

**Issue by Abhijit-2592** [Brophy-E/ECG_GAN_MBD#2]
Hi, thanks for the awesome paper and code. While I was looking through your paper it looks like your Generator's output is within (0,1). But I don't see a Sigmoid activation in the end of the [Generator](https://github.com/Brophy-E/ECG_GAN_MBD/blob/master/Model.py#L172). And you have a comment that states `to make sure the output is between 0 and 1 - removed ,nn.Sigmoid()`. I am kinda confused. Could you throw some light on this? And while training did you use a Sigmoid activation at the end of the Generator?

**Comment by Brophy-E** [comment]
Hi @Abhijit-2592 thank you for your comments and interest in the paper. 

This code should've been double checked again. before upload, we will be uploading a more concise and flexible version in the coming weeks.

In the meantime to answer your query; the output in the paper may appear to be in the bound of (0,1) but if you see Fig. 4, 7 and 8 the bounding is actually in (-1, 1) when `tanh_layer = True`.  So the comment should've read `to make sure the output is between -1 and 1 `. When `tanh_layer = False` the output will be constrained by `nn.Linear` keeping in mind the input data is normalised between 0 & 1.

We did not use Sigmoid activation in the Generator and I note that in the paper it states that sigmoid is used. I will contact the corresponding author to amend this. Thanks.

**Comment by Abhijit-2592** [comment]
@Brophy-E Thanks for the reply.
